src/dial.js

```diff
@@ -97,13 +101,24 @@ module.exports = function dial (swarm) {
           cryptoDial()
 
           function cryptoDial () {
-            // currently, no crypto channel is implemented
             const ms = new multistream.Dialer()
             ms.handle(conn, (err) => {
               if (err) {
                 return cb(err)
               }
-              ms.select('/plaintext/1.0.0', cb)
+
+              const id = swarm._peerInfo.id
+              if (id.privKey == null || swarm.encrypt === false) {
+                return ms.select(tags.plaintex, cb)
+              }
```

Comment:
This should be a Swarm option. In the same fashion that we do .connection.reuse(), we should do .connection.crypto(secio) and then check if it has the ingredients necessary.